src/main/io/lights.c

```diff
@@ -0,0 +1,79 @@
+#include "stdbool.h"
+#include "stdint.h"
+#include "stdlib.h"
+
+#include "platform.h"
+
+#include "common/utils.h"
+
+#include "drivers/time.h"
+/*#include "drivers/io.h"*/
+#include "drivers/lights_hal.h"
+
+#include "rx/rx.h"
+
+#include "fc/config.h"
+#include "fc/rc_controls.h"
+#include "fc/rc_modes.h"
+#include "fc/runtime_config.h"
+
+#include "scheduler/scheduler.h"
+
+#include "config/feature.h"
+
+#include "io/lights.h"
+
+#ifdef USE_LIGHTS
+
+static bool lights_on = false;
+
+#ifdef USE_FAILSAFE_LIGHTS
+  static timeUs_t last_status_change = 0;
+#endif
+
+bool lightsSetStatus(bool status)
+{
+    if (status != lights_on) {
+        lights_on = status;
+        lightsHardwareSetStatus(status);
+        return(true);
+    } else
+        return(false);
+}
+
+/*
+ * Lights handler function to be called periodically in loop. Updates lights
+ * state via time schedule.
+ */
+void lightsUpdate(timeUs_t currentTimeUs)
+{
+    UNUSED(currentTimeUs);
+#ifdef USE_FAILSAFE_LIGHTS
```

Comment:
Why we have `USE_FAILSAFE_LIGHTS`? Why not make it a config option rather than compile-time option?

Reply:
I thought it would not be a setting changed often but you are right it would be better as a run-time setting. I will change that.

Reply:
Done. This PR has been tested and should be ready for merge